numba/cuda/tests/cudapy/test_fastmath.py

```diff
@@ -17,6 +18,93 @@ def foo(arr, val):
         self.assertIn('div.full.ftz.f32', fastver.ptx)
         self.assertNotIn('div.full.ftz.f32', precver.ptx)
 
+    @skip_on_cudasim('fast cos not available in CUDASIM')
+    def test_cosf(self):
+        def f1(r, x):
+            r[0] = cos(x)
+
+        fastver = cuda.jit("void(float32[::1], float32)", fastmath=True)(f1)
+        slowver = cuda.jit("void(float32[::1], float32)")(f1)
+        self.assertIn('cos.approx.ftz.f32 ', fastver.ptx)
+        self.assertNotIn('cos.approx.ftz.f32 ', slowver.ptx)
+
+    @skip_on_cudasim('fast sin not available in CUDASIM')
+    def test_sinf(self):
+        def f2(r, x):
+            r[0] = sin(x)
+
+        fastver = cuda.jit("void(float32[::1], float32)", fastmath=True)(f2)
+        slowver = cuda.jit("void(float32[::1], float32)")(f2)
+        self.assertIn('sin.approx.ftz.f32 ', fastver.ptx)
+        self.assertNotIn('sin.approx.ftz.f32 ', slowver.ptx)
+
+    @skip_on_cudasim('fast tan not available in CUDASIM')
+    def test_tanf(self):
+        def f3(r, x):
+            r[0] = tan(x)
+
+        fastver = cuda.jit("void(float32[::1], float32)", fastmath=True)(f3)
+        slowver = cuda.jit("void(float32[::1], float32)")(f3)
+        self.assertIn('sin.approx.ftz.f32 ', fastver.ptx)
+        self.assertIn('cos.approx.ftz.f32 ', fastver.ptx)
+        self.assertIn('div.approx.ftz.f32 ', fastver.ptx)
+        self.assertNotIn('sin.approx.ftz.f32 ', slowver.ptx)
+
+    @skip_on_cudasim('fast exp not available in CUDASIM')
+    def test_expf(self):
+        def f4(r, x):
+            r[0] = exp(x)
+
+        fastver = cuda.jit("void(float32[::1], float32)", fastmath=True)(f4)
+        slowver = cuda.jit("void(float32[::1], float32)")(f4)
+        self.assertNotIn('fma.rn.f32 ', fastver.ptx)
```

Comment:
Noting here that this checks that the fast version doesn't have any FMA instructions in it. I noticed that the fast version makes use of `ex2.approx.ftz.f32`, which could be checked for instead, but I also think the test as-is is sufficient, because there shouldn't be any FMA instructions in the fast version.

Reply:
@gmarkall I have made and verified the first of the two changes you requested. I am unclear how to make the change to numba/cuda/decorators.py that will update the documentation; specifically I updated 'decorators.py' but after generating the documentation with 'make html', I do not get a updated local page for 'CUDA Kernel API. Is there another command to update this portion of the documentation or did I update the wrong file?

Reply:
I think sphinx isn't very good at detecting when docstrings change. Every time I change something documentation related, I always run `make clean html`, which is a bit inconvenient because it takes a while to build from scratch, but I haven't found a better way.

If you do this, does your change now show up?